cpp/src/arrow/io/interfaces.h

```diff
@@ -51,6 +51,8 @@ struct ReadRange {
 // EXPERIMENTAL
 struct ARROW_EXPORT AsyncContext {
   ::arrow::internal::Executor* executor;
+  // An application-specific ID, forwarded to executor task submissions
+  int64_t external_id = -1;
```

Comment:
@lidavidm Is `int64_t` sufficient here?

Reply:
It should, the other option is to pass a `void *`.